### PR TITLE
Bremsstrahlung

### PR DESCRIPTION
**Pull request by Heikman**
Implement the bremstrahlung effect for electrons. It takes the numerical model from:
 > Gonoskov, A., et al. "Extended particle-in-cell schemes for physics in ultrastrong laser fields: Review and developments." Physical Review E 92.2 (2015): 023305.

**Dependencies:**
 - [ ] #1302 
 - [x] #1299

#### Synchrotron radiation in the classical regime

In a constant B-field the energy loss due to bremsstrahlung is: 
gamma_loss_dgl
Analytical solution:
gamma_loss_analy png
Simulation:
gamma_loss
Zoom:
gamma_loss_zoom
Photon spectrum:
photon_spec_t_0
Zoom:
photon_spec_t_0_zoom

**legend:**
green: analytical result
blue: simulation result, error bars: statistical error

### Known Issues
 - photons are created, but they don't move despite of the photon pusher.

### CODE REVIEW

**Comment by ax3l** [comment]
sweet! :sparkles:
what do the error bars in your plots stand for and which is what (green/blue)?

File: src/picongpu/include/simulation_defines/param/speciesDefinition.param [hunk 1]
```diff
@@ -71,6 +71,29 @@ AttributRadiationFlag
 
 /*########################### define species #################################*/
 
+/*--------------------------- photons -------------------------------------------*/
+
+value_identifier(float_X, DummyMass, 1.0);
```
Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/speciesDefinition.param]: what is the technical problem again with setting mass to zero or not defining it at all?

File: src/picongpu/include/simulation_defines/param/speciesDefinition.param [hunk 2]
```diff
@@ -71,6 +71,29 @@ AttributRadiationFlag
...
+value_identifier(float_X, DummyCharge, 0.0);
```
Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/speciesDefinition.param]: it's actually the charge that is zero, not a dummy mass :)

File: src/picongpu/include/simulation_defines/param/speciesDefinition.param (hunk 1 above)
Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/speciesDefinition.param]: pls call them as usual, they are actually zero and no dummy values.

File: src/picongpu/include/simulation_defines/param/speciesDefinition.param [hunk 3]
```diff
@@ -71,6 +71,29 @@ AttributRadiationFlag
...
+    interpolation<UsedField2Particle>,
+    current<UsedParticleCurrentSolver>,
+    massRatio<DummyMass>,
+    chargeRatio<DummyCharge>
+> ParticleFlagsPhotons;
+
+/*define specie ions*/
+typedef Particles<
+    ParticleDescription<
+        bmpl::string<'p'>,
```
Comment (ax3l) [inline src/picongpu/include/simulation_defines/param/speciesDefinition.param]: Protons would disagree ;)

**Comment by Heikman** [comment]
This PR is aborted because of having too much trouble with micromanagement. Currently the code from this PR is under reconstruction and will be published soon.

**Pull request closed without merging**